source/jormungandr/jormungandr/new_relic.py

```diff
@@ -188,3 +188,39 @@ def record_streetnetwork_call(call_name, connector_name, mode, coverage_name):
 
     # Send the custom event to newrelic !
     record_custom_event(newrelic_service_name, event_params)
+
+
+def statManagerEvent(call_name, group_name):
+    """
+    Custom event that we publish to New Relic for stat_manager
+    """
+
+    def wrap(func):
+        @functools.wraps(func)
+        def wrapper(obj, service, *args, **kwargs):
+            event_params = {
+                "service": type(service).__name__,
+                "call": call_name,
+                "group": group_name,
+                "status": "ok",
+                "az": app.config.get("DEPLOYMENT_AZ", "unknown"),
+            }
+
+            start_time = timeit.default_timer()
+            result = None
+            try:
+                result = func(obj, *args, **kwargs)
+            except Exception as e:
+                event_params["status"] = "failed"
+                event_params.update({"reason": str(e)})
+            duration = timeit.default_timer() - start_time
+            event_params.update({"duration": duration})
+
+            # Send the custom event to newrelic !
+            record_custom_event("stat_manager", event_params)
+
+            return result
```

Comment:
Maybe you should also re-raise in the except, not sure.
More idiomatic:
```suggestion
                return func(obj, *args, **kwargs)
            except Exception as e:
                event_params["status"] = "failed"
                event_params.update({"reason": str(e)})
            finally:
                duration = timeit.default_timer() - start_time
                event_params.update({"duration": duration})

                # Send the custom event to newrelic !
                record_custom_event("stat_manager", event_params)
```

Reply:
Done